django/db/models/sql/compiler.py

```diff
@@ -877,6 +877,9 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                     if self._meta_ordering:
                         order_by = None
                 if having:
+                    if not grouping:
+                        # SQLite does not like HAVING without a GROUP BY
+                        result.append("GROUP BY TRUE")
```

Comment:
We'll see what tests say but this might require a feature flag returning a value to be used in the case of SQLite and set to `None` for other backends.

Particularly curious to see what MySQL will do since `true` is an alias for `1`, due to its lack of proper boolean support, and `GROUP BY`'ing integers has a different semantic.

Reply:
> curious to see what MySQL will do since true is an alias for 1

Looks like you're right! (TIL)